### PR TITLE
Remove Linux + parcon dependency

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Signed-off-by: Alan Jowett <alanjo@microsoft.com>

## Description

Remove dependency on Linux, Python, and parcon for running the bpf2c code gen tests.

## Testing

N/A.

## Documentation

N/A

Resolves: #870
Resolves: #892

### CODE REVIEW

File: tests/bpf2c_tests/bpf_test.cpp [hunk 1]
```diff
@@ -39,16 +39,6 @@ memfrob(uint64_t a, uint64_t b, uint64_t c, uint64_t d, uint64_t e)
 static uint64_t
 trash_registers(uint64_t a, uint64_t b, uint64_t c, uint64_t d, uint64_t e)
```
Comment (dthaler) [inline tests/bpf2c_tests/bpf_test.cpp:40]: Is this function still needed?  It doesn't look like it does what the function is named, with this change.

File: tests/bpf2c_tests/raw_bpf.cpp [hunk 2]
```diff
@@ -117,12 +116,12 @@ run_test(const std::string& data_file)
...
-                                  std::string(prefix) + std::string(".c ") + std::string(" bpf_test.cpp >") +
-                                  std::string(prefix) + std::string(".log -o ") + std::string(prefix) +
-                                  std::string(EXT);
+    std::string compile_command = cc + std::string(" ") + cxxflags +
+                                  std::string(" -I.." SEPERATOR ".." SEPERATOR "include ") + std::string(prefix) +
+                                  std::string(".c ") + std::string(" bpf_test.cpp >") + std::string(prefix) +
+                                  std::string(".log");
     REQUIRE(system(compile_command.c_str()) == 0);
-    std::string test_command = std::string("." SEPERATOR) + std::string(prefix) + std::string(EXT) + std::string(" ") +
+    std::string test_command = std::string("." SEPERATOR) + std::string(prefix) + std::string(" ") +
```
Comment (dthaler) [inline tests/bpf2c_tests/raw_bpf.cpp:124]: "SEPERATOR" is misspelled.  It should be
"SEPARATOR"

**Review by dthaler — DISMISSED** [review]

File: .github/workflows/bpf2c-test.yml [hunk 3]
```diff
@@ -13,49 +13,60 @@ permissions:
...
+        echo "C:\Program Files\OpenCppCoverage" >> $env:GITHUB_PATH
 
-    - name: Restore nuget packages
-      working-directory: ${{env.GITHUB_WORKSPACE}}
+    - name: Set up OpenCppCoverage and add to PATH
+      id: setup_opencppcoverage
       run: |
-        nuget restore
+        choco install -y --requirechecksum=true --checksum=2295A733DA39412C61E4F478677519DD0BB1893D88313CE56B468C9E50517888 --checksum-type=sha256 OpenCppCoverage
+        echo "C:\Program Files\OpenCppCoverage" >> $env:GITHUB_PATH
```
Comment (dthaler) [inline .github/workflows/bpf2c-test.yml:49]: doesn't this duplicate lines 42-43?  Why do you need it in both places?

**Review by dthaler — APPROVED** [review]

File: tests/bpf2c_tests/raw_bpf.cpp [hunk 4]
```diff
@@ -117,20 +116,20 @@ run_test(const std::string& data_file)
...
+    std::string test_command = std::string("." SEPARATOR) + std::string(prefix) + std::string(" ") +
                                std::string(result) + std::string(" \"") + std::string(mem) + std::string("\"");
     REQUIRE(system(test_command.c_str()) == 0);
 }
 
 #define DECLARE_TEST(FILE)                                                                                     \
     TEST_CASE(FILE, "[raw_bpf_code_gen]")                                                                      \
     {                                                                                                          \
-        run_test(".." SEPERATOR ".." SEPERATOR "external" SEPERATOR "ubpf" SEPERATOR "tests" SEPERATOR "" FILE \
+        run_test(".." SEPARATOR ".." SEPARATOR "external" SEPARATOR "ubpf" SEPARATOR "tests" SEPARATOR "" FILE \
```
Comment (shankarseal) [inline tests/bpf2c_tests/raw_bpf.cpp:132]: So this assumes the externel\ubpf directory is in this relative position from the actual test bits. So does this mean this test has to run in the same runner that is executing the build task? Also is it possible for the YML to pass the path to the directory?

**Review by shankarseal — APPROVED** [review]

File: tests/bpf2c_tests/raw_bpf.cpp (hunk 4 above)
Comment (Alan-Jowett) [reply tests/bpf2c_tests/raw_bpf.cpp:132]: Agreed. We should copy the .data files to the output folder and remove the dependency at run time.

**Pull request merged**